docs/content/Cube.js-Introduction.md

```diff
@@ -1,41 +1,75 @@
+  queries into small, fast, reusable and materialized pieces
+- **API Gateway.** Provides an idempotent long polling API as well as a
+  WebSockets API which guarantees delivery of analytical query results without
+  request timeframe limitations and tolerance to connectivity issues
 
 ### Cube.js Frontend
 
-- __Cube.js Javascript Client.__ Core set of methods to access Cube.js API Gateway and to work with query result sets.
-- __Cube.js React, Angular and Vue.__ Framework specific wrappers for Cube.js API.
+- **Javascript Client.** Core SDK for accessing the Cube.js API Gateway and
+  functionality for working with query result sets
+- **React, Angular and Vue.** Framework specific wrappers for Cube.js API
 
 ## Why Cube.js?
 
-If you are building your own business intelligence tool or customer-facing analytics most probably you'll face following problems:
+If you are building your own business intelligence tool or customer-facing
+analytics, it is quite likely you'll face one or more of the following problems:
 
-1. __Performance.__ Most of effort time in modern analytics software development is spent to provide adequate time to insight. In the world where every company data is a big data writing just SQL query to get insight isn't enough anymore.
-2. __SQL code organization.__ Modelling even a dozen of metrics with a dozen of dimensions using pure SQL queries sooner or later becomes a maintenance nightmare which ends up in building modelling framework.
-3. __Infrastructure.__ Key components every production-ready analytics solution requires: analytic SQL generation, query results caching and execution orchestration, data pre-aggregation, security, API for query results fetch, and visualization.
+1. **Performance.** A significant amount of time in modern analytics software
+   development is spent providing adequate time to insight. In a world where
+   every companies' data is big data, writing just SQL queries for insights
+   isn't enough anymore
+2. **SQL code organization.** Modelling even a modest number of metrics and
+   dimensions using pure SQL queries eventually becomes a maintenance nightmare,
+   which then requires engineering effort in building a modelling framework
+3. **Infrastructure.** Any production-ready analytics solution requires key
+   components such as analytic SQL generation, query results caching and
+   execution orchestration, data pre-aggregation, security, API for query
+   results fetch, and visualization
 
-Cube.js has necessary infrastructure for every analytic application that heavily relies on its caching and pre-aggregation layer to provide several minutes raw data to insight delay and sub second API response times on a trillion of data points scale.
+Cube.js has the necessary infrastructure for every analytic application that
+heavily relies on its caching and pre-aggregation layer to provide several
+minutes raw data to insight delay and sub second API response times on a
```

Comment:
I think the word "delay" sounds negative? 
How about "to provide insight from raw data in minutes and sub-second API response times on a trillion data points"?

Reply:
Yeah this whole sentence was a bit iffy ... wasn't entirely sure how to reword it. I like your suggestion, I've slightly added to it:

> to provide insights from raw data within minutes and an API with sub-second response times on up to a trillion data points